client/lib/use-memoized-value/index.ts

```diff
@@ -0,0 +1,18 @@
+import { useRef } from 'react';
+
+export type Predicate< T > = ( prev: T | undefined, next: T ) => boolean;
+
+const defaultCompare = < T >( prev: T | undefined, next: T ): boolean => prev === next;
+
+export function useMemoizedValue< T >(
+	value: T,
+	compare: Predicate< T > = defaultCompare
+): T | undefined {
+	const ref = useRef< T >();
+
+	if ( ! compare( ref.current, value ) ) {
```

Comment:
If the condition was like `ref.current !== undefined && ! compare( ref.current, value )`, then the type of `Predicate`'s `prev` parameter could be just `T`, not `T | undefined`.